sgt-back-end/index.js

```diff
@@ -0,0 +1,159 @@
+      from "grades"
+      `;
+  db.query(sql)
+    .then(result => {
+      const grade = result;
+      if (!grade) {
+        res.status(404).json({
+          Error: 'Cannot find grade'
+        });
+      } else {
+        res.json(grade.rows);
+      }
+
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occured'
+      });
+    });
+
+});
+
+app.post('/api/grades', (req, res) => {
+  const insName = req.body.name;
+  const insCourse = req.body.course;
+  const insScore = req.body.score;
+  if (!insName) {
+    return res.status(400).json({ error: 'Missing input for Name' });
+  } else if (insScore > 100 || insScore < 0) {
+    return res.status(400).json({ error: 'Invalid input for Score' });
+  } else if (!insCourse) {
+    return res.status(400).json({ error: 'Invalid input for Course' });
+  }
+
+  const sql = `
+    insert into "grades" ("name", "course", "score")
+    values ('${insName}', '${insCourse}', '${insScore}')
+    returning*;
+      `;
+  db.query(sql)
+    .then(result => {
+      const grade = result;
+      if (!grade) {
+        res.status(404).json({
+          Error: 'Input Value is incorrect'
+        });
+      } else {
+        res.status(201).json(grade.rows[(grade.rows.length - 1)]);
+      }
+
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occured'
+      });
+    });
+
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const insName = req.body.name;
+  const insCourse = req.body.course;
+  const insScore = req.body.score;
+  const gradeId = req.params.gradeId;
+  if (!insName) {
+    return res.status(400).json({ error: 'Missing input for Name' });
+  } else if (insScore > 100 || insScore < 0) {
+    return res.status(400).json({ error: 'Invalid input for Score' });
+  } else if (!insCourse) {
+    return res.status(400).json({ error: 'Invalid input for Course' });
+  }
+  const sql = `
+    update "grades"
+    set "name" = '${insName}',
+        "course" = '${insCourse}',
+        "score" = '${insScore}'
+    where "gradeId" = ${gradeId}
+    returning*;
+      `;
+  db.query(sql)
+    .then(result => {
+      const grade = result;
+      if (!grade) {
+        res.status(404).json();
+      } else {
+        res.status(201).json(grade.rows);
+      }
+
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occured'
+      });
+    });
+
+});
+
+app.delete('/api/grades/:gradeId', (req, res) => {
+
+  const gradeId = req.params.gradeId;
+
+  if (req.params.gradeId !== gradeId) {
+    res.status(404).json({ error: 'The target grade does not exist in the database' });
+
+  }
+
+  const sql = `
+    delete from "grades"
+    where "gradeId" = ${gradeId}
+    returning*;
+      `;
+  db.query(sql)
+    .then(result => {
+      const grade = result;
+      if (gradeId in grade.rows) {
```

Comment:
This is where you find out if `gradeId` is valid or not, so if you set `grade` from `rows[0]` you can check if it exists or not, if the id was bad you won't have any data.
```JAVASCRIPT
const grade = result.rows[0];

if (!grade) {
  // No grade was found, send an error response
}
```